### PR TITLE
feat(jupyterhub): unify logs configuration for better log management

### LINKED ISSUE #123: jupyterhub: unify logs configuration for better log management

### PR DESCRIPTION
**Pull request by mehdibn**
#### Which issue(s) this PR fixes

Fixes #123

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: roles/jupyterhub/server/templates/jupyterhub_config.py.j2 [hunk 1]
```diff
@@ -130,3 +130,18 @@ def pre_spawn_hook(spawner):
 {% for key, value in jupyterhub_properties.get('yarnspawner', {}).items() +%}
 c.YarnSpawner.{{ key }} = {% if value is string and not value.startswith('f:') %}'{{ value }}'{% else %}{{ value if value is not string else value.lstrip('f:') }}{% endif %}
 {% endfor %}
+
```
Comment (kpgtek) [inline roles/jupyterhub/server/templates/jupyterhub_config.py.j2:133]: Perharps here needs few line breaks to separate logging configuration section.

File: roles/jupyterhub/server/templates/jupyterhub_start.sh.j2 [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-{{ jupyterhub_install_dir }}/bin/python -m jupyterhub.app -f {{ jupyterhub_server_conf_dir }}/jupyterhub_config.py &>> {{ jupyterhub_log_dir }}/jupyterhub.log
+{{ jupyterhub_install_dir }}/bin/python -m jupyterhub.app -f {{ jupyterhub_server_conf_dir }}/jupyterhub_config.py {% if jupyterhub_root_logger != "RFA" and jupyterhub_root_logger != "DRFA"%}&>> {{ jupyterhub_log_dir }}/{{ jupyterhub_log_file }}{% endif %}
```
Comment (kpgtek) [inline roles/jupyterhub/server/templates/jupyterhub_start.sh.j2:3]: Space character missing at the end of the condition ```{% if condition %}```.

File: roles/jupyterhub/server/templates/jupyterhub_config.py.j2 (hunk 1 above)
Comment (mehdibn) [reply roles/jupyterhub/server/templates/jupyterhub_config.py.j2:133]: i kept the same existing config, but, we can set `INFO` as Hue

File: roles/jupyterhub/server/templates/jupyterhub_start.sh.j2 (hunk 2 above)
Comment (mehdibn) [reply roles/jupyterhub/server/templates/jupyterhub_start.sh.j2:3]: thanks !

File: tdp_vars_defaults/jupyterhub/jupyterhub.yml [hunk 3]
```diff
@@ -111,3 +108,22 @@ jupyterhub_start_on_boot: no
 
 # Service restart policies
 jupyterhub_restart: "no"
+
+# Jupyterhub logging configuration
+# Root logger level should be: [FATAL| ERROR| WARN| INFO| DEBUG| TRACE]
+jupyterhub_log_dir: /var/log/jupyterhub
+jupyterhub_log_file: "jupyterhub_server_{{ ansible_fqdn }}.log"
+jupyterhub_root_logger_level: DEBUG
```
[the remark this answers is not in the document]
Comment (mehdibn) [reply tdp_vars_defaults/jupyterhub/jupyterhub.yml:116]: I kept the existing config, but, we can set `INFO` as Hue

File: roles/jupyterhub/server/templates/jupyterhub_config.py.j2 (hunk 1 above)
Comment (mehdibn) [reply roles/jupyterhub/server/templates/jupyterhub_config.py.j2:133]: thanks !

**Comment by gonzaloetjo** [comment]
I tested the deployment and it worked correctly with jupyer. I'll wait for the changes @kpgtek asked to validate.

File: tdp_vars_defaults/jupyterhub/jupyterhub.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mehdibn) [reply tdp_vars_defaults/jupyterhub/jupyterhub.yml]: done

File: roles/jupyterhub/server/templates/jupyterhub_start.sh.j2 (hunk 2 above)
Comment (mehdibn) [reply roles/jupyterhub/server/templates/jupyterhub_start.sh.j2]: done

**Comment by mehdibn** [comment]
> I tested the deployment and it worked correctly with jupyer. I'll wait for the changes @kpgtek asked to validate.

done

**Review by kpgtek — APPROVED** [review]

**Review by rpignolet — APPROVED** [review]

**Review by gonzaloetjo — APPROVED** [review]

**Pull request merged**